src/components/crownBoundaryEventDropdown.vue

```diff
@@ -77,6 +81,16 @@ export default {
 
       this.$emit('add-boundary-event', node);
     },
+    tooltip(nodeType) {
+      if (getEmptyBoundaryEventPositionsForShape(this.shape).length === 0) {
+        return 'No available space';
+      }
+      if (!canAddBoundaryEventToTarget(nodeType, this.shape)
+        && nodeType === 'processmaker-modeler-boundary-error-event') {
+        return 'Can only add one per Task';
+      }
+      return null;
```

Comment:
This feels like a polymorphic structure trying to escape. :) Could this be extracted using the 'replace conditional with polymorphism' refactoring?

Reply:
I might need help with that.